src/core.jl

```diff
@@ -28,30 +84,112 @@ function show(io::IO, e::Edge)
     print(io, "edge $(e.first) - $(e.second)")
 end
 
-"""A type representing an undirected graph."""
-type Graph
-    vertices::UnitRange{Int}
+type SparseGraph<:AbstractSparseGraph
     edges::Set{Edge}
-    fadjlist::Vector{Vector{Int}} # [src]: (dst, dst, dst)
-    badjlist::Vector{Vector{Int}} # [dst]: (src, src, src)
+    fm::SparseMatrixCSC{Bool, Int}
 end
 
-"""A type representing a directed graph."""
-type DiGraph
-    vertices::UnitRange{Int}
+type SparseDiGraph<:AbstractSparseGraph
     edges::Set{Edge}
-    fadjlist::Vector{Vector{Int}} # [src]: (dst, dst, dst)
-    badjlist::Vector{Vector{Int}} # [dst]: (src, src, src)
+    fm::SparseMatrixCSC{Bool, Int}
```

Comment:
Primarily because it's very efficient to look up forward and backward adjacencies this way. Consider an edge (1,2). It will have fadjlist[1] = [2, ...] and badjlist[2] = [1, ...].

Note also that we're changing the game a bit with the move to sparse matrices for version 0.4. (Check out the 'sparsemx' branch for details). Directed Graphs will have two sparse matrices (accessible via `fmat` and `bmat`); undirected only needs one.